hawkular-inventory-api/src/main/java/org/hawkular/inventory/base/BaseTenants.java

```diff
@@ -46,7 +46,7 @@ private BaseTenants() {
 
     }
 
-    public static class ReadWrite<BE> extends Mutator<BE, Tenant, Tenant.Blueprint, Tenant.Update>
+    public static class ReadWrite<BE> extends Mutator<BE, Tenant, Tenant.Blueprint, Tenant.Update, String>
```

Comment:
The reason for adding this is that configs don't have any IDs so they use `Void` as their id type. The configs are identified by the role. Which brings me to a possible refactoring:

Instead of `configuration` and `connectionConfiguration` methods in `Resources.Single|Multiple` we could use the `DataEntity.Role` as the id and have:
`Resources.Single#data(DataEntity.Role)` method. This would IMHO be maybe better than the current approach of a method per data role.